### PR TITLE
Test.Tasty.Ingredients: add composeReporters

### PR DESCRIPTION
**Pull request by minad**
as discussed in #158

### CODE REVIEW

File: core/Test/Tasty/Ingredients.hs [hunk 1]
```diff
@@ -119,3 +121,21 @@ suiteOptions ins tree =
...
+-- it only works for 'TestReporter' ingredients.
+composeReporters :: Ingredient -> Ingredient -> Ingredient
+composeReporters (TestReporter o1 f1) (TestReporter o2 f2) =
+  TestReporter (o1 ++ o2) $ \o t ->
+  case (f1 o t, f2 o t) of
+    (g, Nothing) -> g
+    (Nothing, g) -> g
+    (Just g1, Just g2) -> Just $ \s -> do
+      (h1, h2) <- concurrently (g1 s) (g2 s)
+      return $ \x -> fmap fst $ concurrently (h1 x) (h2 x)
```
Comment (feuerbach) [inline core/Test/Tasty/Ingredients.hs]: I'd rather have a `&&` of the two results here just to be safe.

Comment (minad) [inline core/Test/Tasty/Ingredients.hs]: I agree!

**Comment by feuerbach** [comment]
Great — could you add this to your original commit and force-push the branch? See https://ro-che.info/articles/2016-09-18-good-pull-requests

**Comment by minad** [comment]
force pushed

**Comment by feuerbach** [comment]
Awesome, thanks.

**Comment by minad** [comment]
Thank you! What are your plans for a new minor release? Do you want to get in other stuff or fixes first or are you ok with a new minor release for just this function? I would love to use it in tasty-stats :) With respect to tasty-stats a minor release is ok if I edit the upper bounds on hackage.

**Comment by feuerbach** [comment]
Sure, I've just pushed 0.11.2 to hackage.

**Comment by minad** [comment]
oh, that was fast! thank you so much!

**Pull request merged**